### ISSUE TITLE
Add API for prefetching extension information

### ISSUE THREAD

**Issue by psychon** [psychon/x11rb#200]
It should be possible for API users to avoid blocking here:
https://github.com/psychon/x11rb/blob/a1deb7e2940ce3c2c06650c7a03bddc15a250423/src/extension_information.rs#L26-L27

**Comment by psychon** [comment]
After having attempted this: The problem is here:
```rust
pub fn query_extension<'c, Conn>(conn: &'c Conn, name: &[u8]) -> Result<Cookie<'c, Conn, QueryExtensionReply>, ConnectionError>
where Conn: RequestConnection + ?Sized
```
The `Cookie` contains a reference to the connection and thus needs its lifetime. This means that the cookie cannot be saved inside of the connection, because this ends up creating a self-referential struct.

**Comment by psychon** [comment]
Fixed by 1238da8a26eb0bae71b87eca7b6b9329a7a5e509
